open-company/open-company-web
# Local env fixes

Problem: when we dev for mobile we change `localhost` with the our local network ip address in the `local-settings` file to be able to point our real devices to the dev server.
That requires to keep some unstashed changes in the local_settings.cljs file that could be committed by accident.
With this change `oc.web.local-settings` reads the some vars from a js file loaded only in local dev which may or may not contain the values. If the file doesn't exist or a certain var is not included the default are used.

To test:
- make sure you don't have a file in `resources/public/js/local-env.js`
- load [https://localhost:3559](https://localhost:3559)
- everything works fine?
- do you see a 404 for the `/js/local-env.js` file (this is the only negative effect but it's harmless)
- now create a file in `resources/public/js/local-env.js`
- set the local-env.js file content to something like the example in the README.md file
- now load the marketing site using your ip address
- did it work?
- make sure staging branch has this code merge in
- deploy it
- staging working fine?

NB: if you want to use the app you have to change the hostname also in Storage and Auth (at least).
